aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/AWSS3StoragePlugin.java

```diff
@@ -121,11 +129,16 @@ public final class AWSS3StoragePlugin extends StoragePlugin<S3Client> {
     private final ExecutorService executorService;
     private final AuthCredentialsProvider authCredentialsProvider;
     private final AWSS3StoragePluginConfiguration awsS3StoragePluginConfiguration;
-    private AWSS3StorageService storageService;
+    private AWSS3StorageService defaultStorageService;
     @SuppressWarnings("deprecation")
     private StorageAccessLevel defaultAccessLevel;
     private int defaultUrlExpiration;
 
+    private Map<String, AWSS3StorageService> awsS3StorageServicesByBucketName;
```

Comment:
What if 2 buckets have same name but different region? I think the key here should be ResolvedStorageBucket.

Reply:
Can we also add the default storage service in this map so that we don't end up with a duplicate if customer explicitly requests the default bucket.

Reply:
Yes, the default storage service is always added by default at configure time [see line 290](https://github.com/aws-amplify/amplify-android/pull/2895/files#diff-68c08b683554135bdca3ec8741ef1160af2a1add61a87d302af6719c214b46cdR290)

Bucket names are unique globally across all regions in S3; so I think it's appropriate to use it as the key here.  It's not possible to have two buckets with the same name but different region.